specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/WebApps.json

```diff
@@ -29260,6 +29669,116 @@
         "modelAsString": true
       }
     },
+    "WorkflowEnvelope": {
+      "description": "Workflow properties definition.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The resource id."
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Gets the resource name."
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Gets the resource type."
+        },
+        "kind": {
+          "type": "string",
```

Comment:
Can you use commontypes to define the resource ?

Reply:
Can you give an example of what you're looking for? These single string values are defined like this everywhere else in our definition.

Reply:
Bump, any links to what you're requesting?

Reply:
I assume raosuhas is thinking of something along the lines of this:

>Instead of defining your own resource class with required top level properties, 'id', 'type', 'name', and 'location' and 'tags' (for TrackedResource) its recommended to inherit from ProxyResource or TrackedResource in common-types v2 or v3.json. (The v2 definition does NOT include systemData, and is otherwise similar to the v3 one).
[https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#L56](https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#L56)